app/components/Home/WalletLauncher.js

```diff
@@ -69,66 +129,135 @@ class WalletLauncher extends React.Component {
-
-        {wallet.type === 'local' && (
-          <>
-            <Heading.h1 mb={5}>
-              <FormattedMessage {...messages.settings_title} />
-            </Heading.h1>
-
-            <WalletSettingsFormLocal
-              key={wallet.id}
-              id={`wallet-settings-form-${wallet.id}`}
-              wallet={wallet}
-              startLnd={startLnd}
-            />
-          </>
-        )}
-
-        {wallet.type !== 'local' && (
-          <>
-            <WalletSettingsFormRemote
-              key={wallet.id}
-              id={`wallet-settings-form-${wallet.id}`}
-              wallet={wallet}
-              startLnd={startLnd}
-            />
-          </>
-        )}
-
-        <Text mt={4} fontWeight="normal">
-          <FormattedMessage {...messages.section_delete_title} />
-        </Text>
-        <Bar my={2} />
-
-        <Flex justifyContent="center" my={4}>
-          <Button size="small" onClick={this.handleDelete}>
-            <FormattedMessage {...messages.delete_wallet_button_text} />
-          </Button>
-        </Flex>
-      </React.Fragment>
+      <Box css={{ height: '100%', 'overflow-y': 'overlay' }} pt={4} px={5} pb={6}>
+        <Form getApi={this.setFormApi} onSubmit={this.onSubmit} initialValues={walletConverted}>
+          {() => (
+            <>
+              <Flex mb={4} alignItems="center">
+                <Box width="75%" mr={3}>
+                  <WalletHeader wallet={wallet} />
+                </Box>
+                <Flex ml="auto" justifyContent="flex-end" flexDirection="column">
+                  <Button type="submit" size="small" ml={2}>
+                    <FormattedMessage {...messages.launch_wallet_button_text} />
+                  </Button>
+                </Flex>
+              </Flex>
+
+              {wallet.type === 'local' ? (
+                <>
+                  <Heading.h1 mb={5}>
+                    <FormattedMessage {...messages.settings_title} />
+                  </Heading.h1>
+
+                  <WalletSettingsFormLocal
+                    autopilotDefaults={autopilotDefaults}
+                    wallet={walletConverted}
+                  />
+                </>
+              ) : (
+                <WalletSettingsFormRemote wallet={walletConverted} />
+              )}
+
+              <Text mt={4} fontWeight="normal">
+                <FormattedMessage {...messages.section_delete_title} />
+              </Text>
+              <Bar my={2} />
+
+              <Flex justifyContent="center" my={4}>
+                <Button size="small" onClick={this.handleDelete} type="button">
+                  <FormattedMessage {...messages.delete_wallet_button_text} />
+                </Button>
+              </Flex>
+
+              {this.hasChanges() && <WalletActionBar buttons={actionBarButtons} />}
+            </>
+          )}
+        </Form>
+      </Box>
     )
   }
 }
 
-export default withRouter(WalletLauncher)
+export default compose(
```

Comment:
Nice. I never knew about this

Reply:
Makes it a lot easier to comprehend rather than `hoc1(hoc2(hoc3(cmp)))` imo